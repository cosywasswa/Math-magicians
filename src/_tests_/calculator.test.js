import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../components/calculator';

describe('Calculator component', () => {
  const props = {
    total: 0,
    operation: null,
    next: null,
  };
  it('should render calculator elements ', () => {
    const { getByTestId } = render(<Calculator />);
    const output = getByTestId('calcDiv');
    expect(output).toBeTruthy();
  });

  it('should render next prop elements as null', () => {
    render(<Calculator props={props} />);
    const input = screen.getByTestId('row');
    expect(input).toBeInTheDocument;
    expect(props.next).toBe(null);
  });

  it('should render correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON;
    expect(tree).toMatchSnapshot();
  });
});
