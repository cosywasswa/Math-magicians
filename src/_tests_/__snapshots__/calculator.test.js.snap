// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component should render correctly 1`] = `[Function]`;

exports[`Calculator component should render next prop elements as null 1`] = `[Function]`;
